datastore.go

```diff
@@ -79,15 +79,25 @@ var DefaultOptions Options
 
 func init() {
 	DefaultOptions = Options{
-		GcDiscardRatio: 0.2,
+		GcDiscardRatio: 0.01,
 		GcInterval:     15 * time.Minute,
 		GcSleep:        10 * time.Second,
 		Options:        badger.LSMOnlyOptions(""),
 	}
-	// This is to optimize the database on close so it can be opened
-	// read-only and efficiently queried. We don't do that and hanging on
-	// stop isn't nice.
-	DefaultOptions.Options.CompactL0OnClose = false
+	// This is to optimize the database on closure
+	DefaultOptions.Options.CompactL0OnClose = true
+	
+	// Remove elements which the has been deleted from the database
+	DefaultOptions.Options.NumVersionsToKeep = 0
+	
+	// Reduce the number of zero tables (which are hold in memory)
+	DefaultOptions.Options.NumLevelZeroTables = 1
+	
+	// Reduce the number of zero tables which are stalled
+	DefaultOptions.Options.NumLevelZeroTablesStall = 2
+	
+	// Reduce the max vlog size usage after compaction (
+	DefaultOptions.Options.ValueLogFileSize = 10485760
```

Comment:
This sets the maximum value log file size to 10 MiB. That will definitely help fix the issue, however:

1. It will set a maximum value size of 10MiB (probably fine but I'm not sure).
2. I believe badger keeps value logs open. That will mean a 100x increase in the number of file descriptors use by badger which won't work for us (20GiB -> 2048 file descriptors versus 20 now).

Reply:
As far as I understand this option, badger won't try to compact value logs when they are below ValueLogFileSize. So it the GC will only reduce the ValueLogFileSize when it's above this value.

Maybe @jarifibrahim can help us out a little and clear this up?

Reply:
Hey @RubenKelevra let me try to explain. 
The value log file is the write ahead log file for badger. All the new writes will be added to this file. A smaller vlog file size is recommended only if you'd like value log GC to clean up disk space easily. 
Note - Compaction is for SST files. GC is for value log file. 

As a result of this change, you will have too many value log files which means too many open file descriptors. I do not suggest settings value log file size to such a small value. If your value log file is 10mb, badger will not be able to store values more than 9 mb (there are headers and checksums as well). 

If 1 gb seems like too big for your usecase, I suggest using 500 mb and evaluate if the new change is an improvement or not. Dropping file size to 10 mb could lead to strange issues (I've never seen anyone use a 10 mb file size. That's too low)
I haven't gone through the entire change in this PR and the issue but are you guys having issues with the GC? If this is something new can you please create a ticket on badger so that I can figure out the right fix for your issue?

Reply:
Please feel free to ping me @jarifibrahim if something doesn't make sense. The last thing we would want is setting incorrect options for badger :)